### PR TITLE
Happy Thoughts - Karolin Albinsson

### PR DESCRIPTION
**Pull request by karolinalbinsson**
Link to my deployed project:
https://karolin-happy-thoughts.netlify.app/ 

My reflections on how this week's project turned out:
I think it turned out good. I aimed for black level.

Things I'd like to have clarified or explained in more detail:
-

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,93 @@
...
+          <Thought 
+          key= {thought._id}
+            id = {thought._id}
+            message = {thought.message}
+            createdAt = {thought.createdAt}
+            hearts = {thought.hearts}
+            onLike = {handleLikes}
+          />
+          ))}
+    </section>}
```
Comment (ylvan75) [inline code/src/App.js:90]: Here you could have used ternary operator (if I'm thinking correct now ;-)) 

`{loading ? (<Loader>) : (<section ...)`

File: code/src/Form.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          What's making you happy right now?
+            <textarea 
+            className="text-input" 
+            rows="5" 
+            value={thoughtText} 
+            required 
+            onChange={(event) => handleChange(event)}></textarea>
+          </label>
+          {inputLengthOK && <p className="input-ok">{numChars}/140</p>}
+          {!inputLengthOK && <p className="input-not-ok">{numChars}/140</p>}
```
Comment (ylvan75) [inline code/src/Form.js:41]: Here you could have used ternary operator (if I'm thinking correct now ;-))

`{inputLengthOK ? ( <p className="input-ok">{numChars}/140</p>) : (<p className="input-not-ok">{numChars}/140</p>)`

File: code/src/index.css [hunk 3]
```diff
@@ -1,13 +1,214 @@
...
+.header {
+	height: 49px;
+	width: 100%;
+	position: sticky;
+	top: 0;
+	background: rgb(255, 255, 255);
+	background: linear-gradient(
+		180deg,
+		rgba(255, 255, 255, 1) 65%,
+		rgba(9, 9, 121, 0) 100%
```
Comment (ylvan75) [inline code/src/index.css:33]: On Safari - The sticky navbar gets a  blue shadow at the bottom with this code.

**Review by ylvan75 — APPROVED** [review]
Really nice work Karro ❤️ 

Good and clear README. Like that it is short and concise.

I like that you have a loader and also nice with the animation when posting a new happy thought.


I took some time and looked at your site with these browsers/devices and it worked fine. I observed two things:
1) On Safari - The sticky navbar has a blue shadow at the bottom. I saw it when I tested on iPhone (have Safari as default browser). I made a comment in index.css line 29-33.
2) On Chrome (Mac) and Edge (Windows)- It's only at theses browser that I get a good indication that the "heart button" is disabled after I have liked something. Doesn't look as good in Firefox or iPhone (Safari) or Samsung (Chrome)  

Cross-browser
-Chrome - Mac
-Firefox - Windows
-Edge - Windows
-iOS - iPone X (Safari)
-Android - Samsung S8 (Chrome)

Don't forget to be consistent if you want to use semi-colon.

I have added some comments but nothing that needs to be fixed.

File: code/src/App.js [hunk 4]
```diff
@@ -1,9 +1,93 @@
-import React from 'react'
+import React, { useState, useEffect } from "react";
+import { LikeCounter } from "LikeCounter";
+import { Loader } from "Loader";
+import { Thought } from "Thought";
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/App.js:4]: 👍

File: code/src/App.js [hunk 5]
```diff
@@ -1,9 +1,93 @@
...
+  }
+
+  const handleSubmit = (newMessage) => {
+    postThought(newMessage);
+  }
+
+  const postThought = (message) => {
+    const myHeaders = new Headers();
+    myHeaders.append("Content-Type", "application/json");
+    const messageToPost = JSON.stringify({"message": message});
```
Comment (EmOlsz) [inline code/src/App.js:42]: left-side `message` shouldn't be wrapped inside quotes, it's supposed to be key of the object, and they are not strings or anything else, just names. Needs to be fixed

File: code/src/App.js [hunk 6]
```diff
@@ -1,9 +1,93 @@
...
+    fetch("https://happy-thoughts-technigo.herokuapp.com/thoughts", requestOptions)
+    .then((res) => 
+      res.json()
+    )
+    .then((newThought) => { 
+      console.log(newThought);
+      if(newThought.errors !== undefined)
+        {
+          setMessageOK(false);
+          console.log(newThought.message);
```
Comment (EmOlsz) [inline code/src/App.js:59]: `console.error()` would be better for that

File: code/src/Thought.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+  return(
+  <div className="thought-card">
+    <p className="message">{props.message}</p>
+    <div className="bottom-container">
+   
+    {!liked && 
+    <div>
+      <button 
+      className={numLikes === 0 ?  "no-likes-button" : (numLikes > 0 && numLikes < 3 ? "like-button" : "mega-like-button")}
```
Comment (EmOlsz) [inline code/src/Thought.js:45]: Those indentations could be improved ;)

**Comment by karolinalbinsson** [comment]
Thanks for the feedback. I have made the requested changes.

**Pull request closed without merging**